File: python/primes/play.py
"""prime
gerar um array de primes de 1 até 1000
"""

def is_prime(p):
    for i in range(2,int(p/2)+1):
        if p % i == 0:
            return False
    return True

def primes(n):
    return list(filter(is_prime,range(1,n)))

print(primes(1000))

File: ruby/primes/play.rb
def is_prime(n)
  return false if n < 1
  return true if n < 3
  return false if n.even?

  for i in (3..n/2).filter { |e| e.odd? } do
    return false if n % i == 0
  end

  true
end

def primes(n)
  (1..n).filter {|e| is_prime(e)}
end

primes(1000).each { |e| puts e }
